qiskit/circuit/measure.py

```diff
@@ -59,3 +61,53 @@ def measure(self, qubit, cbit):
 
 
 QuantumCircuit.measure = measure
+
+
+class PauliMeasure(Measure):
+    """Perform a measurement with preceding basis change operations."""
+
+    def __init__(self, basis='z'):
```

Comment:
Can you use capital 'X', 'Y', 'Z' here to match labels used for Pauli's in the `Pauli` class and other locations.

Also I wouldn't make this a kwarg with a default. I would make it required arg.